### DISCUSSION TITLE
Feature Request: VM-based Android integration

### DISCUSSION THREAD

**Discussion by upintheairsheep** [waydroid/waydroid#1059] [Ideas]
The majority of all issues faced on Waydroid are using the alternative to virtualizating Android, and it prevents Waydroid from being ported to other platforms such as Windows or macOS, therefore I suggest to implement an alternate solution that silently runs a modified build of Android in QEMU and passes the input to various integrations such as window integration.

**Comment by ChrysoliteAzalea** [comment]
You can try running BlissOS in a virtual machine.

**Reply by upintheairsheep** [reply]
Ruins the whole point. Windows cannot be directly side by side with native apps.

**Reply by Xtr126** [reply]
> Ruins the whole point. Windows cannot be directly side by side with native apps.

Still technically possible if you manage to build Bliss OS with waydroid wayland hwcomposer and use virtio-wayland / virtio-gpu with crosvm to connect to host wayland server. Running waydroid in a vm would be bloat since it requires two OS running in the VM, the linux distro and Android container, which would take more disk space and memory. Instead of just Android.

**Reply by upintheairsheep** [reply]
> > Ruins the whole point. Windows cannot be directly side by side with native apps.
> 
> Still technically possible if you manage to build Bliss OS with waydroid wayland hwcomposer and use virtio-wayland / virtio-gpu with crosvm to connect to host wayland server. Running waydroid in a vm would be bloat since it requires two OS running in the VM, the linux distro and Android container, which would take more disk space and memory. Instead of just Android.

Was saying windows as in boxes of apps, not the OS. I want seamless Android apps, I would have used a normal VM otherwise

**Reply by upintheairsheep** [reply]
> > Ruins the whole point. Windows cannot be directly side by side with native apps.
> 
> Still technically possible if you manage to build Bliss OS with waydroid wayland hwcomposer and use virtio-wayland / virtio-gpu with crosvm to connect to host wayland server. Running waydroid in a vm would be bloat since it requires two OS running in the VM, the linux distro and Android container, which would take more disk space and memory. Instead of just Android.

ChromeOS does exactly that. It runs Play Store apps in a VM, while it takes more disk space and RAM, it's usable. The nature of Linux compared to ChromeOS makes this viable, as most Linux PCs can easily run it.

**Comment by upintheairsheep** [comment]
I’m saying to run regular Android in a VM, with bliss OS, but use the
waydroid bridges.

**Reply by Shaarigan** [reply]
Use a regular Android Emulator for Windows and MAC. Switching the entire backend just to support other OS is nuts and against the initial idea of AnBox/Waydroid using kernel similarities between Android/Linux in order to speed the entire emulation process up and beat regular emulation software. Looking at the docs it says AnBox/Waydroid use Linux containers, a virtualization technique especially present in Linux.

This might be done in MAC as well with some effort since both kernels are based on UNIX Kernel but Windows is out of reach as they developed their own closed-source kernel no one else uses these days and it won't have any benefit against established emulation software specialized to Android or some generic ones like VirtualBox or VM Ware

**Comment by kairusds** [comment]
crosvm already does that, though you'd need to compile it yourself to suit your needs

**Reply by upintheairsheep** [reply]
arc only supports ChromeOS, unless someone has a port.

**Comment by upintheairsheep** [comment]
Anyone ported arcvm to “normal” Linux?

**Comment by Pallas13** [comment]
I extracted the Android Arc image and its kernel.  But have problems in selinux permission and cgroup problem.
